### PR TITLE
Implemented parsing of keep statements and struct variables

### PR DESCRIPTION
**Pull request by XiaoFei9704**
#### Description
Implemented parsing of keep statements and struct variables.
Just like:
`# 设置速度
    speed_a: velocity
    keep (default speed_a == 30kph)`
`struct weather:
    fog: fog   
    rain: precipitation
struct precipitation:
    intensity: velocity 
struct fog:
    visual_range: length 
`
`# 设置天气
    w: weather
    keep(w.fog.visual_range == 150m) # 能见度
    keep(w.rain.intensity == 10mps) # 降水量`
Now the system can correctly recognize the meaning of this sentence. Next, we can call Carla's API to extend the Domain model.
#### Where has this been tested?
  * **Platform(s):** Ubuntu 20.04
  * **Python version(s):** Python 3.8.10
  * **Unreal Engine version(s):** UE4
  * **CARLA version:** 0.9.13

### CODE REVIEW

**Comment by XiaoFei9704** [comment]
Mainly modified in osc2_scenario.py file

File: srunner/scenarioconfigs/osc2_scenario_configuration.py [hunk 1]
```diff
@@ -169,20 +161,19 @@ def visit_scenario_declaration(self,  node: ast_node.ScenarioDeclaration):
                 elif isinstance(child, ast_node.DoDirective):
                     self.visit_do_directive(child)
 
-        def visit_do_directive(self,  node: ast_node.DoDirective):
-            #场景设置信息不会出现在场景的行为描述部分
+        def visit_do_directive(self, node: ast_node.DoDirective):
+            # 场景设置信息不会出现在场景的行为描述部分
```
Comment (charles-lei) [inline srunner/scenarioconfigs/osc2_scenario_configuration.py:165]: Is this function useless?

File: srunner/scenarioconfigs/osc2_scenario_configuration.py [hunk 2]
```diff
@@ -312,22 +303,22 @@ def visit_event_declaration(self, node: ast_node.EventDeclaration):
...
                     self.visit_binary_expression(child)
-        
+
         def visit_binary_expression(self, node: ast_node.BinaryExpression):
             arguments = [self.visit_children(node), node.operator]
             flat_arguments = flat_list(arguments)
             temp_stack = []
             for ex in flat_arguments:
-                if ex == '+' or ex == '-' or ex == '*' or ex == '/' or ex == '%' :
+                if ex == '+' or ex == '-' or ex == '*' or ex == '/' or ex == '%':
```
Comment (charles-lei) [inline srunner/scenarioconfigs/osc2_scenario_configuration.py:321]: how about following code:
`if ex in ['+', '-', '*', '/', '%']`

**Review by charles-lei — APPROVED** [review]
Look good to me!

File: srunner/scenarioconfigs/osc2_scenario_configuration.py (hunk 1 above)
Comment (XiaoFei9704) [reply srunner/scenarioconfigs/osc2_scenario_configuration.py:165]: It's necessary. Its function is to execute the statement following the 'do' keyword.

File: srunner/scenarioconfigs/osc2_scenario_configuration.py (hunk 2 above)
Comment (XiaoFei9704) [reply srunner/scenarioconfigs/osc2_scenario_configuration.py:321]: > how about following code: `if ex in ['+', '-', '*', '/', '%']`
OK, it's an optimal solution.

**Pull request merged**